### PR TITLE
Add information about `key` to `matrix` docs

### PR DESCRIPTION
**Pull request by pbielak**
Docs update for https://github.com/iterative/dvc/issues/10043

### CODE REVIEW

File: content/docs/user-guide/project-structure/dvcyaml-files.md [hunk 1]
```diff
@@ -725,28 +725,30 @@ stages:
       feature: [feature1, feature2, feature3]
     cmd: ./train.py --feature ${item.feature} ${item.model}
     outs:
-      - ${item.model}.pkl
+      - ${key}.pkl # equivalent to: ${item.model}-${item.feature}.pkl
```
Comment (dberenbaum) [inline content/docs/user-guide/project-structure/dvcyaml-files.md:728]: Not a docs question, but does the core PR guarantee the ordering of keys (model first here) matches the way it's written in `dvc.yaml`?

File: content/docs/user-guide/project-structure/dvcyaml-files.md [hunk 2]
```diff
@@ -725,28 +725,30 @@ stages:
...
 ```
 
 You can reference each variable in your stage definition using the `item`
 dictionary key. In the above example, you can access `item.model` and
-`item.feature`.
+`item.feature`. Moreover, `matrix` exposes a `key` value, which combines the
+current `item` values into one expression (equal to the current stage name after
+the `@` – see example below).
```
Comment (dberenbaum) [inline content/docs/user-guide/project-structure/dvcyaml-files.md:735]:
```suggestion
current `item` values into one expression (see example below).
```

You don't have to accept this suggestion, but I think this part is more confusing than helpful -- seeing the example is more clear than explaining it.

**Review by dberenbaum — APPROVED** [review]
LGTM, thanks @pbielak!

File: content/docs/user-guide/project-structure/dvcyaml-files.md (hunk 1 above)
Comment (pbielak) [reply content/docs/user-guide/project-structure/dvcyaml-files.md:728]: As Python dicts keep the ordering, the generated key should reflect the order of the matrix inputs: https://github.com/iterative/dvc/blob/main/dvc/parsing/__init__.py#L533

**Pull request merged**